WordPress/Sniff.php

```diff
@@ -550,6 +550,76 @@ public function strip_quotes( $string ) {
 	}
 
 	/**
+	 * Add a PHPCS message to the output stack as either a warning or an error.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param string $message   The message.
+	 * @param int    $stackPtr  The position of the token the message relates to.
+	 * @param bool   $is_error  Optional. Whether to report the message as an 'error' or 'warning'.
+	 *                          Defaults to true (error).
+	 * @param string $code      Optional error code for the message. Defaults to 'Found'.
+	 * @param array  $data      Optional input for the data replacements.
+	 * @param int    $severity  Optional. Severity level. Defaults to 0 which will translate to
+	 *                          the PHPCS default severity level.
+	 * @return bool
+	 */
+	protected function addMessage( $message, $stackPtr, $is_error = true, $code = 'Found', $data = array(), $severity = 0 ) {
```

Comment:
> Instead of $is_error being a boolean, should it perhaps be a string error or warning? Just thinking that in the future I would hope PHPCS would add an informational message that isn't error or warning. 

I considered that, but AFAICS PHPCS does not have any plans to add more message types in the (near) future. I didn't see anything about it in the changelogs or plans for v3 either if I remember correctly.

The drawback of changing `$is_error` to a string is that this would add some small overhead each time the function is called as you would either need a ternary `( true === $something ) ? 'error' : 'warning'` of a full if/else conditional in which you set an `$errortype` variable you would then pass to the function.
The boolean just keeps things simpler in that regards.

> Also, passing 'error' is more readable than passing true.

True and I know WP is a big fan of "wordy" variables. But - unlike WP with themes and plugins -, I don't expect non-programmers to start writing code for WPCS. The method is well documented, so for any developer, I would expect this to be quite intuïtive.
Or am I overlooking something here ?